packages/samples/headless-ssr/src/app/page.tsx

```diff
@@ -0,0 +1,13 @@
+import {fetchInitialState} from '@/src/common/engine';
+import SearchPage from '@/src/components/search-page';
+import {Suspense} from 'react';
+
+// Entry point SSR function
+export default async function Search() {
+  const engineSnapshot = await fetchInitialState({controllers: {}});
+  return (
+    <Suspense fallback={<div>Hydrating engine ..</div>}>
+      <SearchPage engineSnapshot={engineSnapshot}></SearchPage>
+    </Suspense>
```

Comment:
Looks like this isn't working. When I checkout your branch, the server-side-rendered page is blank.

I believe the reason is because Suspense only works with some APIs (see the "Note" below https://react.dev/reference/react/Suspense#displaying-a-fallback-while-content-is-loading). It also works with the `use` hook, but [it's not recommended by Next.js yet](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-client).

Reply:
Anyway, this is a pretty React-specific concept, and I think we should demonstrate using the snapshot's state as well as the hydrated state.